tests/test_zero/low_level_zero/test_zero1_2.py

```diff
@@ -106,31 +107,34 @@ def check_sharded_optim_against_torch_ddp():
     We feed these two sets of models with the same input and check if the
     differences in model output and updated parameters are within tolerance.
     """
+    local_rank = torch.distributed.get_rank()
+    seed_all(1453)
 
-    # create layer
-    zero_linear1 = nn.Linear(128, 256)
-    zero_linear2 = nn.Linear(256, 512)
-
-    # create model
-    zero_model = nn.Sequential(zero_linear1, zero_linear2)
+    # create models
+    zero_model = TestModel()
     torch_model = copy.deepcopy(zero_model)
 
     zero_model = zero_model.cuda().half()
-    torch_model = DDP(torch_model.cuda())
+    # torch_model = DDP(torch_model.cuda(), bucket_cap_mb=0)
+    torch_model = torch_model.cuda()
+
+    # for (n, p), z1p in zip(torch_model.named_parameters(), zero_model.parameters()):
+    #    half_close(p.data, z1p.data)
 
     # create optimizer
-    zero_optimizer = torch.optim.Adam(zero_model.parameters(), lr=0.001)
+    zero_optimizer = torch.optim.SGD(zero_model.parameters(), lr=1)
```

Comment:
what if optimizer is Adam? will it still be OK? I belive ZeRO is exactly proposed for the 2nd order optimizers.

Reply:
Is Adam ready to use in this PR?

Reply:
Adam is used in this pr, see file `test_grad_acc.py`. but it is not ready for `[test_zero1_2.py](https://github.com/hpcaitech/ColossalAI/pull/1964/files/d99723ec6dc109b662a1b83691f1ee9603fb450a#diff-3cc07fc076702467742b211fc8b5d8283a7f5d2cf80c6b6fe42b98be3d6cee66)`